### PR TITLE
Converter: Fix 'bytes' field decoding into an empty array, now decoded into an empty Buffer

### PR DESCRIPTION
**Pull request by assaf-xm**

### CODE REVIEW

**Comment by assaf-xm** [comment]
Should fix https://github.com/dcodeIO/protobuf.js/issues/1012

**Comment by assaf-xm** [comment]
@dcodeIO can you take a look on the fix?

File: src/converter.js [hunk 1]
```diff
@@ -245,7 +245,7 @@ converter.toObject = function toObject(mtype) {
         ("}else")
             ("d%s=o.longs===String?%j:%i", prop, field.typeDefault.toString(), field.typeDefault.toNumber());
             else if (field.bytes) gen
-        ("d%s=o.bytes===String?%j:%s", prop, String.fromCharCode.apply(String, field.typeDefault), "[" + Array.prototype.slice.call(field.typeDefault).join(",") + "]");
+        ("d%s=o.bytes===String?%j:o.bytes===Array?%s:%s", prop, String.fromCharCode.apply(String, field.typeDefault), "[" + Array.prototype.slice.call(field.typeDefault).join(",") + "]", "util.newBuffer([])");
```
[the remark this answers is not in the document]
Comment (assaf-xm) [reply src/converter.js]: Pushed a fix

[the remark this answers is not in the document]
Comment (dcodeIO) [reply src/converter.js]: Hmm, this'd still generate the same array literal two times, right?

[the remark this answers is not in the document]
Comment (assaf-xm) [reply src/converter.js]: It looks like:
object.buf = options.bytes === String ? "" : options.bytes === Array ? [] : $util.newBuffer([]);

[the remark this answers is not in the document]
Comment (dcodeIO) [reply src/converter.js]: Yeah, if the default value is empty. What's on my mind is something along the lines of:

```js
if (options.bytes === String) object.buf = "...";
else {
  object.buf = [...];
  if (options.bytes !== Array) object.buf = $util.newBuffer(object.buf);
}
```

[the remark this answers is not in the document]
Comment (assaf-xm) [reply src/converter.js]: OK possible as well, pushed now.
The generated code is:
```javascript
    if (options.bytes === String)
        object.buf = "";
    else {
        object.buf = [];
        if (options.bytes !== Array)
            object.buf = $util.newBuffer(object.buf);
        }
```

[the remark this answers is not in the document]
Comment (assaf-xm) [reply src/converter.js]: @dcodeIO , is it better now?

**Comment by assaf-xm** [comment]
@dcodeIO , the PR is ready for a second review.

**Comment by dcodeIO** [comment]
Thanks for the heads up! Looking good to me now :)

File: src/converter.js [hunk 2]
```diff
@@ -244,9 +244,15 @@ converter.toObject = function toObject(mtype) {
             ("d%s=o.longs===String?n.toString():o.longs===Number?n.toNumber():n", prop)
         ("}else")
             ("d%s=o.longs===String?%j:%i", prop, field.typeDefault.toString(), field.typeDefault.toNumber());
-            else if (field.bytes) gen
-        ("d%s=o.bytes===String?%j:%s", prop, String.fromCharCode.apply(String, field.typeDefault), "[" + Array.prototype.slice.call(field.typeDefault).join(",") + "]");
-            else gen
+            else if (field.bytes) {
+                var arrayDefault = "[" + Array.prototype.slice.call(field.typeDefault).join(",") + "]";
+                gen
+        ("if(o.bytes===String) d%s=%j", prop, String.fromCharCode.apply(String, field.typeDefault))
```
Comment (dcodeIO) [inline src/converter.js]: For example, can remove whitespace at `) d`

File: src/converter.js [hunk 3]
```diff
@@ -244,9 +244,15 @@ converter.toObject = function toObject(mtype) {
...
+        ("else {")
```
Comment (dcodeIO) [inline src/converter.js]: at `e {`

File: src/converter.js [hunk 4]
```diff
@@ -244,9 +244,15 @@ converter.toObject = function toObject(mtype) {
...
+            ("d%s=%s", prop, arrayDefault)
+            ("if(o.bytes!==Array) d%s=util.newBuffer(d%s)", prop, prop)
```
Comment (dcodeIO) [inline src/converter.js]: at `) d`

**Comment by assaf-xm** [comment]
100%, whitespaces were removed

**Comment by assaf-xm** [comment]
@dcodeIO - I hope it's ready now

**Comment by dcodeIO** [comment]
Thank you! :)

**Comment by assaf-xm** [comment]
@dcodeIO , Hi, when the next release including this fix is expected?

**Comment by dcodeIO** [comment]
There are some build issues with node v10 currently that I haven't yet been able to investigate.

**Comment by assaf-xm** [comment]
@dcodeIO , when do you expect the next release to be issued?

**Pull request merged**